controllers/policyserver_controller_test.go

```diff
@@ -142,59 +146,333 @@ var _ = Describe("PolicyServer controller", func() {
+	})
+
+	It("should creates the PolicyServer pod with the limits and the requests", func() {
+		policyServer := policyServerFactory(policyServerName)
+		policyServer.Spec.Limits = corev1.ResourceList{
+			"cpu":    resource.MustParse("100m"),
+			"memory": resource.MustParse("1Gi"),
+		}
+		createPolicyServerAndWaitForItsService(policyServer)
+		By("creating a deployment with limits and requests set")
+		Eventually(func() error {
+			deployment, err := getTestPolicyServerDeployment(policyServerName)
+			if err != nil {
 				return err
-			}, timeout, pollInterval).Should(Succeed())
-			configmap, err := getTestPolicyServerConfigMap(policyServerName)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(configmap).To(PointTo(MatchFields(IgnoreExtras, Fields{
-				"Data": MatchAllKeys(Keys{
-					constants.PolicyServerConfigPoliciesEntry: Equal("{}"),
-					constants.PolicyServerConfigSourcesEntry:  Equal("{}"),
-				}),
-			})))
-		})
+			}
+			Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Limits).To(Equal(policyServer.Spec.Limits))
+			return nil
+		}, timeout, pollInterval).Should(Succeed())
+
+		By("creating a pod with limit and request set")
+		Eventually(func() error {
+			pod, err := getTestPolicyServerPod(policyServerName)
+			if err != nil {
+				return err
+			}
 
-		It("it should create a PDB when policy server is defined with MinAvailable", func() {
-			minAvailable := intstr.FromInt(2)
-			policyServer.Spec.MinAvailable = &minAvailable
-			createPolicyServerAndWaitForItsService(policyServer)
-			Eventually(func() *k8spoliciesv1.PodDisruptionBudget {
-				pdb, _ := getPolicyServerPodDisruptionBudget(policyServerName)
-				return pdb
-			}, timeout, pollInterval).Should(policyServerPodDisruptionBudgetMatcher(policyServer, &minAvailable, nil))
-		})
+			Expect(pod.Spec.Containers[0].Resources.Limits).To(Equal(policyServer.Spec.Limits))
 
-		It("should create a PodDisruptionBudget when policy server is defined with MaxUnavailable", func() {
-			maxUnavailable := intstr.FromInt(2)
-			policyServer.Spec.MaxUnavailable = &maxUnavailable
-			createPolicyServerAndWaitForItsService(policyServer)
-			Eventually(func() *k8spoliciesv1.PodDisruptionBudget {
-				pdb, _ := getPolicyServerPodDisruptionBudget(policyServerName)
-				return pdb
-			}, timeout, pollInterval).Should(policyServerPodDisruptionBudgetMatcher(policyServer, nil, &maxUnavailable))
-		})
+			By("setting the requests to the same value as the limits")
+			Expect(pod.Spec.Containers[0].Resources.Requests).To(Equal(policyServer.Spec.Limits))
 
-		It("it should not create PDB when policy server is defined with no PodDisruptionBudget", func() {
-			createPolicyServerAndWaitForItsService(policyServer)
-			Consistently(func() error {
-				_, err := getPolicyServerPodDisruptionBudget(policyServerName)
+			return nil
+		}, timeout, pollInterval).Should(Succeed())
+	})
+
+	It("when the requests are updated should update the PolicyServer pod with the new requests", func() {
```

Comment:
maybe rephrase (It)"should update the PolicyServer pod with the new requests when the requests are updated"